lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	ManifestRepo string
+	MasterSHA    string
+}
+
+// Run executes ImageRemovalCheck on a set of promotion edges.
+func (check ImageRemovalCheck) Run(
+	edges map[PromotionEdge]interface{},
+) error {
+	var err error
+	// Clone the manifest dir into a temporary directory
+	tempDir, err := ioutil.TempDir("", "k8s.io-")
+	if err != nil {
+		return fmt.Errorf("Could not create temporary Git repo %v: %v",
+			tempDir, err)
+	}
+	_, err = gogit.PlainClone(tempDir, false, &gogit.CloneOptions{
+		URL: check.ManifestRepo,
+	})
+	if err != nil {
+		return fmt.Errorf("Could not clone Git repo: %v",
+			err)
+	}
+
+	// Parse the manifests and generate the promotion edges
+	mfests, _ := ParseThinManifestsFromDir(tempDir)
+	masterEdges, _ := ToPromotionEdges(mfests)
+
+	destinationImages := make(map[ImageTag]interface{})
+	for edge := range edges {
+		destinationImages[edge.DstImageTag] = nil
+	}
+
+	// Check that every destination image in the master set of promotion edges
+	// exists in the pull request set of promotion edges
+	removedImages := make([]string, 0)
+	for edge := range masterEdges {
+		_, found := destinationImages[edge.DstImageTag]
+		if !found {
+			removedImages = append(removedImages,
+				string(edge.DstImageTag.ImageName))
+		}
+	}
+
+	// Remove the temporary directory
```

Comment:
As stated above, since you don't need to clone, you don't need this either. In general though, you would have probably used a `defer` here to do this instead of calling this directly in-between the removal check logic.